Rubberduck.VBEEditor/Events/VBENativeServices.cs

```diff
@@ -207,15 +235,23 @@ private static ICodePane GetCodePaneFromHwnd(IntPtr hwnd)
 
         private static IWindow GetWindowFromHwnd(IntPtr hwnd)
         {
-            if (!User32.IsWindow(hwnd))
+            if (!User32.IsWindow(hwnd) || _vbe == null)
             {
                 return null;
             }
 
             var caption = hwnd.GetWindowText();
             using (var windows = _vbe.Windows)
             {
-                return windows.FirstOrDefault(x => x.Caption.Equals(caption));
+                foreach (var window in windows)
+                {
+                    if (window.Caption.Equals(caption))
+                    {
+                        return window;
+                    }
+                    window.Dispose();
```

Comment:
Same concern RE: exceptions causing leaks because `window.Dispose()` won't be `finally`'d.

Reply:
As above, if you put the dispose in the finally block, you dispose the return value.

Reply:
I think I have an idea how to handle this. However, that does not make the code prettier.